### PR TITLE
Quick review

### PR DESCRIPTION
**Pull request by jrisebor**
This is just a first pass, thoughts?

### CODE REVIEW

File: hotkeys.py [hunk 1]
```diff
@@ -58,37 +52,34 @@ def get_context(app):
...
         context = "🤷🏻‍♂️"
-        if proc.returncode == 0:
-            process_lines = out.split("\n")
-            # S+ or R+
-            # TODO parse the process line, then check for the flag
-            processes = [p for p in process_lines if "S+" in p]
-            if len(processes) == 0:
-                processes = [p for p in process_lines if "R+" in p]
+        process_lines = out.split("\n")
+        # S+ or R+
```
Comment (jrisebor) [inline hotkeys.py:58]: After reading the `man` page for `ps`, I think we might just be able to parse for `+` to find the process we are looking for 🤔

Comment (donaldducky) [reply hotkeys.py:58]: I think in some cases, I saw two `S+` processes.

Using a `make` command that performs other commands generates multiple `S+` processes.
I was originally using `+` and got `S+` and `R+` at one point.

File: hotkeys.py [hunk 2]
```diff
@@ -58,37 +52,34 @@ def get_context(app):
...
-                        proc = subprocess.Popen(lsof_cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, universal_newlines=True)
-                        out, err = proc.communicate()
-                        try:
-                            cwd = [c for c in out.split("\n") if re.match(r'^n', c)][0][1:]
-                        except IndexError:
-                            print(f"Could not find cwd from {out}")
+                    lsof_cmd = ["lsof", "-a", "-p", match['pid'], "-d", "cwd", "-F", "n"]
+                    out = subprocess.run(lsof_cmd, capture_output=True, universal_newlines=True, check=True).stdout
+                    try:
+                        cwd = [c for c in out.split("\n") if c.startswith("n")][0][1:]
```
Comment (donaldducky) [inline hotkeys.py:75]: 😂 I need to learn some basic python skills here.

File: hotkeys.py [hunk 3]
```diff
@@ -97,17 +88,23 @@ def get_context(app):
...
+            active_app = NSWorkspace.sharedWorkspace().activeApplication()
+            context = get_context(active_app)
+            if context['id'] != 'hotkeys' and (context['id'] != last_active_id or context['cwd'] != last_cwd):
+                last_active_id = context['id']
+                last_cwd = context['cwd']
+                on_context_change(context)
+            sleep(1)
+        except KeyboardInterrupt:
+            print(f'{SHOW_CURSOR}')
+            sys.exit(0)
```
Comment (donaldducky) [inline hotkeys.py:107]: `ctrl+c` is the normal case.

Is there a good way to detect 'on exit script'?

I was initially using signal handling to catch `SIGINT` but I would like to be able to handle `SIGTERM` and crashes as well.

**Review by donaldducky — CHANGES_REQUESTED** [review]
I'm wondering if we could figure out a better way to handle exiting the script.

That's the only thing I'm a bit iffy about atm.

File: hotkeys.py (hunk 1 above)
Comment (donaldducky) [reply hotkeys.py:58]: Here's one example, using `aws ec2 help`:
```
Found an unexpected number of processes [
'82125 S+   /usr/local/Cellar/python@3.9/3.9.2_1/Frameworks/Python.framework/Versions/3.9/Resources/Python.app/Contents/MacOS/Python /usr/local/bin/aws ec2 help',
'82131 S+   less -R'
]
```

Comment (donaldducky) [reply hotkeys.py:58]: Maybe we should split this discussion into an issue.

File: hotkeys.py (hunk 3 above)
Comment (donaldducky) [reply hotkeys.py:107]: Created #3 for this.

File: hotkeys.py (hunk 1 above)
Comment (donaldducky) [reply hotkeys.py:58]: Created #2

**Pull request merged**